### PR TITLE
Fix windows paths in test/run.d

### PR DESCRIPTION
**Pull request by marler8997**

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: test/run.d [hunk 1]
```diff
@@ -141,7 +142,7 @@ auto predefinedTargets(string[] targets)
 {
     static findFiles(string dir)
     {
-        return scriptDir.buildPath(dir).dirEntries("*{.d,.sh}", SpanMode.shallow).map!(e => e.name);
+        return dirEntries(testPath(dir), "*{.d,.sh}", SpanMode.shallow).map!(e => e.name);
```
Comment (wilzbach) [inline test/run.d]: Why do you change the style? UFCS is the Phobos style.

File: test/run.d [hunk 2]
```diff
@@ -185,7 +186,7 @@ auto filterTargets(string[] targets, string[string] env)
     bool error;
     foreach (target; targets)
     {
-        if (!scriptDir.buildPath(target).exists)
+        if (!exists(testPath(target)))
```
Comment (wilzbach) [inline test/run.d]: Why do you change the style? UFCS is the Phobos style.

File: test/run.d [hunk 3]
```diff
@@ -16,7 +16,8 @@ import std.algorithm, std.conv, std.datetime, std.exception, std.file, std.forma
 import core.stdc.stdlib : exit;
 
 const scriptDir = __FILE_FULL_PATH__.dirName.buildNormalizedPath;
-string resultsDir = scriptDir.buildPath("test_results");
+auto testPath(R)(R path) { return buildNormalizedPath(scriptDir, path); }
```
Comment (wilzbach) [inline test/run.d]: scriptDir is already normalized in the line above.

File: test/run.d [hunk 4]
```diff
@@ -260,17 +261,17 @@ string[string] getEnvironment()
...
         env["LIB"] = phobosPath;
 
         // auto-tester might run the testsuite with a different $(MODEL) than DMD
         // has been compiled with. Hence we manually check which binary exists.
         // For windows the $(OS) during build is: `windows`
-        int dmdModel = "../generated/windows/%s/64/dmd%s".format(build, exe).exists ? 64 : 32;
+        int dmdModel = testPath(format(`..\generated\windows\%s\64\dmd%s`, build, exe)).exists ? 64 : 32;
         env.getDefault("MODEL", dmdModel.text);
-        env["DMD"] = "../generated/windows/%s/%d/dmd%s".format(build, dmdModel, exe);
+        env["DMD"] = testPath(format(`..\generated\windows\%s\%d\dmd%s`, build, dmdModel, exe));
```
Comment (wilzbach) [inline test/run.d]: Looks like these changes are the only ones needed (this was just copied/translated from the makefile which still assumes that it's called from a posix-like shell, e.g. like the git shell on Windows).

**Review by wilzbach — COMMENTED** [review]
Thanks a lot, but I'm not sure whether we need all the changes.

File: test/run.d (hunk 3 above)
Comment (marler8997) [reply test/run.d]: Right, but `path` may not be normalized.

File: test/run.d (hunk 1 above)
Comment (marler8997) [reply test/run.d]: What criteria should we use to determine when and when not to use UFCS?

**Comment by marler8997** [comment]
@wilzbach I reverted the UFCS style changes in the lines that were modified.

**Review by wilzbach — APPROVED** [review]

File: test/run.d (hunk 1 above)
Comment (wilzbach) [reply test/run.d]: Not really a role, but for things like `exists` it just reads nicer and more like English, so imho there wasn't a good reason to revert it.

**Pull request merged**